src/lib/Microsoft.Health.Fhir.R4.Ingest/Service/R4FhirImportService.cs

```diff
@@ -76,6 +76,12 @@ public virtual async Task<string> SaveObservationAsync(ILookupTemplate<IFhirTemp
             if (!_observationCache.TryGetValue(cacheKey, out Model.Observation existingObservation))
             {
                 existingObservation = await GetObservationFromServerAsync(identifier).ConfigureAwait(false);
+
+                // Discovered an issue where FHIR Service is only matching on first 128 characters of the identifier.  This is a temporary measure to prevent merging of different observations until a fix is available.
+                if (existingObservation != null && !existingObservation.Identifier.Exists(i => i.IsExactly(identifier)))
+                {
+                    throw new NotSupportedException("FHIR Service returned matching observation but expected identifier was not present.");

```

Comment:
nit: Just curious if we are allowed to include the identifier in the exception message?

Reply:
It could be PII so no.